tests/test_app.py

```diff
@@ -287,6 +287,44 @@ def test_session_stats_without_provider_id(self):
         session = Session.query.get('123')
         self.assertIsNone(session)
 
+    def test_session_stats_without_bytes_sent_received(self):
+        payload = {
+            'bytes_received': 40,
+            'provider_id': '0x1',
+        }
+        auth = generate_test_authorization(json.dumps(payload))
+        re = self._post(
+            '/v1/sessions/123/stats',
+            payload,
+            headers=auth['headers'],
+        )
+
+        self.assertEqual(400, re.status_code)
+        self.assertEqual(
+            {'error': 'bytes_sent missing or value is not unsigned int'},
+            re.json
+        )
+
+        payload = {
+            'bytes_sent': 20,
+            'provider_id': '0x1',
+        }
+        auth = generate_test_authorization(json.dumps(payload))
+        re = self._post(
+            '/v1/sessions/123/stats',
+            payload,
+            headers=auth['headers'],
+        )
+
+        self.assertEqual(400, re.status_code)
+        self.assertEqual(
+            {'error': 'bytes_received missing or value is not unsigned int'},
+            re.json
+        )
+
+        sessions = Session.query.all()
+        self.assertEqual(0, len(sessions))
```

Comment:
This should be 2 separate tests IMO - without_bytes_sent and without_bytes_received. This would make tests isolated - each test tests exactly one case. Test results would be more helpful, i.e. you would see which attribute fails and which passes in case only one fails.

Reply:
improved